base/util.jl

```diff
@@ -750,3 +750,47 @@ kwdef_val(::Type{Cwstring}) = Cwstring(C_NULL)
 kwdef_val{T<:Integer}(::Type{T}) = zero(T)
 
 kwdef_val{T}(::Type{T}) = T()
+
+"""
+    @callsuper f(x::T, y::S, ...)
+
+Call the method of function `f` with parameter signature specified by `T`, `S`, etc.
+"""
+macro callsuper(ex)
+    ex.head == :call || error("@invoke requires a call expression")
+    fname = ex.args[1]
+    args = ex.args[2:end]
+    types = Symbol[]
+    vals = Symbol[]
+    kwargs = Any[]
+    blk = quote end
+    for arg in args
+        if isa(arg,Expr) && arg.head == :kw
```

Comment:
@yuyichao, wouldn't be easier all around if you just did this PR? There was a similar back and forth when I proposed a possible implementation